### PR TITLE
Modified to Restpy

### PR DESCRIPTION
**Pull request by desaimg1**

### CODE REVIEW

File: Modules/IxNetRestApiPortMgmt.py [hunk 1]
```diff
@@ -13,28 +16,24 @@ def setMainObject(self, mainObject):
...
-            self.ixnObj.logInfo('connectToVChassis error: %s' % response.json()['result'])
-            return 1
-        else:
-            self.ixnObj.logInfo('connectToVChassis: Successfully connected to chassis: %s' % chassisIp)
-            while response.json()["state"] == "IN_PROGRESS" or response.json()["state"] == "down":
-                if timeout == 0:
-                    break
+        counterStop = 45
+        for counter in range(1, counterStop + 1):
+            # chassisStatus = self.ixNetObj.AvailableHardware.add_Chassis(Hostname=ixChassisIp)
```
Comment (ANISH-GOTTAPU) [inline Modules/IxNetRestApiPortMgmt.py:24]: why commented part is added here, remove it

File: Modules/IxNetRestApiPortMgmt.py [hunk 2]
```diff
@@ -13,28 +16,24 @@ def setMainObject(self, mainObject):
         # For Python Robot Framework support
         self.ixnObj = mainObject
 
-    def connectToVChassis(self, chassisIp):
+    def connectToVChassis(self, ixChassisIp):
```
Comment (keysightgems) [inline Modules/IxNetRestApiPortMgmt.py:19]: Why argument name modified from chassisIp to ixChassisIp, do not modify argument names.

File: Modules/IxNetRestApiPortMgmt.py [hunk 3]
```diff
@@ -13,28 +16,24 @@ def setMainObject(self, mainObject):
...
-        if response == 1: return 1
-        if response.json()['state'] == 'ERROR':
-            self.ixnObj.logInfo('connectToVChassis error: %s' % response.json()['result'])
-            return 1
-        else:
-            self.ixnObj.logInfo('connectToVChassis: Successfully connected to chassis: %s' % chassisIp)
-            while response.json()["state"] == "IN_PROGRESS" or response.json()["state"] == "down":
-                if timeout == 0:
-                    break
+        counterStop = 45
```
Comment (keysightgems) [inline Modules/IxNetRestApiPortMgmt.py:22]: This parameter not required, you can give value direct in for loop.
Remove and modify the statements like below (To optimize/reduce execution time).
for counter in range(1, 45):
       .....
       if chassisStatus.State != 'ready':
       .....
       if chassisStatus.State == 'ready':
       .....
       if counter > 45:
       .....

File: Modules/IxNetRestApiPortMgmt.py [hunk 4]
```diff
@@ -57,47 +56,24 @@ def connectIxChassis(self, chassisIp, timeout=30, **kwargs):
...
-            # Chassis states: down, polling, ready
-
-            for timer in range(1,timeout+1):
-                response = self.ixnObj.get(self.ixnObj.httpHeader + chassisIdObj, silentMode=True)
-                currentStatus = response.json()['state']
-                self.ixnObj.logInfo('connectIxChassis {0}: Status: {1}. Wait {2}/{3} seconds'.format(
-                    chassisIpAddress, currentStatus, timer, timeout), timestamp=False)
-
-                if currentStatus != 'ready' and timer < timeout:
+        counterStop = timeout
```
Comment (keysightgems) [inline Modules/IxNetRestApiPortMgmt.py:59]: Why creating new argument here, use "timeout" argument directly.

File: Modules/IxNetRestApiPortMgmt.py [hunk 5]
```diff
@@ -128,10 +104,7 @@ def getChassisId(self, chassisIp):
         Parameter
            chassisIp: <str>: The chassis IP address
         """
-        response = self.ixnObj.get(self.ixnObj.sessionUrl+'/availableHardware/chassis')
-        for eachChassis in response.json():
-            if eachChassis['ip'] == chassisIp:
-                return eachChassis['id']
+        pass
```
Comment (keysightgems) [inline Modules/IxNetRestApiPortMgmt.py:107]: We don't have id in Restpy but we have the option to get the Id using a href like below. Please check with Anish once and implement stead of the "Pass" statement.
Example: href = /api/v1/sessions/1/ixnetwork/availableHardware/chassis/1
                Id = 1
               href = /api/v1/sessions/1/ixnetwork/availableHardware/chassis/2
                Id = 2

File: Modules/IxNetRestApiPortMgmt.py [hunk 6]
```diff
@@ -181,28 +143,10 @@ def createVports(self, portList=None, rawTrafficVportStyle=False):
...
-                response = self.ixnObj.patch(self.ixnObj.httpHeader+vportObj, data={'name': portNumber})
 
-        if createdVportList == []:
-            raise IxNetRestApiException('No vports created')
-
-        self.ixnObj.logInfo('createVports: %s' % createdVportList)
-        return createdVportList
+        self.ixnObj.logInfo('\n Creating Vports for portList {}'.format(portList))
+        for i in range(1,len(portList)+1) :
+            self.ixNetObj.Vport.add()
```
Comment (keysightgems) [inline Modules/IxNetRestApiPortMgmt.py:149]: Missed return statement, add.
Missed to handle rawTrafficVportStyle parameter.

File: Modules/IxNetRestApiPortMgmt.py [hunk 7]
```diff
@@ -212,13 +156,15 @@ def getVportObjectByName(self, portName):
...
-            response = self.ixnObj.get(vportObj)
-            if response.json()['name'] == portName:
-                return vportObj
-        return None
+
+        self.ixnObj.logInfo("getting vport object for portname {}".format(portName))
+        for vport in self.ixNetObj.Vport.find() :
+            if vport.Name == portName :
+                return vport
+                break
```
Comment (keysightgems) [inline Modules/IxNetRestApiPortMgmt.py:164]: break not required here, remove

File: Modules/IxNetRestApiPortMgmt.py [hunk 8]
```diff
@@ -240,8 +186,13 @@ def linkUpDown(self, port, action='down'):
...
             action: <str>: up|down
         """
-        vport = self.ixnObj.getVports([port])[0]
-        self.ixnObj.post(self.ixnObj.sessionUrl+'/vport/operations/linkUpDn', data={'arg1': [vport], 'arg2': action})
+
+        action = action.lower()
+        self.ixnObj.logInfo("Link {} operation for ports {}".format(action,port))
+        for eachPort in port:
+            self.ixnObj.logInfo("\n Make port {} to {} state".format(":".join(eachPort),action))
+            for vport in self.ixNetObj.Vport.find(AssignedTo=":".join(eachPort)):
```
Comment (keysightgems) [inline Modules/IxNetRestApiPortMgmt.py:194]: Here "for" loop not required use like below.
self.ixNetObj.Vport.find(AssignedTo=":".join(eachPort))

File: Modules/IxNetRestApiPortMgmt.py [hunk 9]
```diff
@@ -251,8 +202,8 @@ def getAllVportList(self):
         Returns
             List of vports: ['/api/v1/sessions/1/ixnetwork/vport/1', '/api/v1/sessions/1/ixnetwork/vport/2']
         """
-        response = self.ixnObj.get(self.ixnObj.sessionUrl+'/vport')
-        vportList = ['%s' % vport['links'][0]['href'] for vport in response.json()]
+        self.ixnObj.logInfo("Get vports for all ports")
+        vportList = [vport for vport in self.ixNetObj.Vport.find()]
```
Comment (keysightgems) [inline Modules/IxNetRestApiPortMgmt.py:206]: Here list comprehensive not required, use direct "find" method to return list of vports.

File: Modules/IxNetRestApiPortMgmt.py [hunk 10]
```diff
@@ -381,108 +331,19 @@ def assignPorts(self, portList, forceTakePortOwnership=True, createVports=False,
...
+        self.ixnObj.logInfo('\nAssignPorts: {0}'.format(portList))
 
-        # Verify if there is existing vports. If yes, user either loaded a saved config file or 
-        # the configuration already has vports.
-        # If loading a saved config file and reassigning ports, assign ports to existing vports.
-        response = self.ixnObj.get(self.ixnObj.sessionUrl+'/vport')
+        try :
+            self.ixNetObj.AssignPorts(testPorts, [], vportList, forceTakePortOwnership)
+        except :
+            raise Exception("Failed to Assign ports {} ".format(portList))
```
Comment (keysightgems) [inline Modules/IxNetRestApiPortMgmt.py:345]: Why not handled rawTraffic, configPortName parameters, those are not required in Restpy?
Missed return statement here, add

File: Modules/IxNetRestApiPortMgmt.py [hunk 11]
```diff
@@ -528,16 +385,22 @@ def releasePorts(self, portList):
...
-            if vport == []:
-                continue
-            vportList.append(vport[0])
 
-        url = self.ixnObj.sessionUrl+'/vport/operations/releaseport'
-        response = self.ixnObj.post(url, data={'arg1': vportList})
-        self.ixnObj.waitForComplete(response, url+'/'+response.json()['id'], timeout=120)
+        self.ixnObj.logInfo('Release selected ports {} from configuration'.format(portList))
+        vportList=self.getVports(portList)
+        counterStop = 30
```
Comment (keysightgems) [inline Modules/IxNetRestApiPortMgmt.py:391]: Here counterStop variable not using, remove

File: Modules/IxNetRestApiPortMgmt.py [hunk 12]
```diff
@@ -528,16 +385,22 @@ def releasePorts(self, portList):
...
+        for vport in vportList :
+            vport.ReleasePort()
+            # for counter in range(1,counterStop+1) :
```
Comment (keysightgems) [inline Modules/IxNetRestApiPortMgmt.py:394]: remove commented lines.

File: Modules/IxNetRestApiPortMgmt.py [hunk 13]
```diff
@@ -563,20 +427,13 @@ def clearPortOwnership(self, portList):
...
-                userPortId = userPort[2]
-                url = self.ixnObj.httpHeader+chassisHref+'/card/'+str(userCardId)+'/port/'+str(userPortId)+'/operations/clearownership'
-                data = {'arg1': [chassisHref+'/card/'+str(userCardId)+'/port/'+str(userPortId)]}
-                self.ixnObj.post(url, data=data)
+        self.ixnObj.logInfo('Clear port ownership for ports {}'.format(portList))
+        for port in portList:
+            ixChassisIp = port[0]
+            cardId = port[1]
+            portId = port[2]
+            portIdObj = self.ixNetObj.AvailableHardware.Chassis.find(Ip=ixChassisIp)[0].Card.find(CardId=cardId)[0].Port.find(PortId=portId)[0]
```
Comment (keysightgems) [inline Modules/IxNetRestApiPortMgmt.py:435]: Here checking chassisIp/card/port only with zero index, handle other index as well.

File: Modules/IxNetRestApiPortMgmt.py [hunk 14]
```diff
@@ -631,42 +485,26 @@ def arePortsAvailable(self, portList, raiseException=True):
...
-        self.ixnObj.logInfo('Ports are still owned: %s' % portOwnedList)
-
-        if portOwnedList != []:
-            if raiseException:
-                raise IxNetRestApiException('arePortsAvailable: Ports are still owned')
-            else:
-                return portOwnedList
-        return 0
+
+            portIdObj = self.ixNetObj.AvailableHardware.Chassis.find(Ip=ixChassisIp)[0].Card.find(CardId=cardId)[0].Port.find(PortId=portId)[0]
```
Comment (keysightgems) [inline Modules/IxNetRestApiPortMgmt.py:494]: Here checking chassisIp/card/port only with zero index, handle other index as well.

File: Modules/IxNetRestApiPortMgmt.py [hunk 15]
```diff
@@ -838,13 +672,7 @@ def setTxMode(self, vportList='all', txMode='interleaved', timeout=70):
...
            timeout:   <int>: the timeout value to declare as failed. Default=70 seconds.
         """
-        if vportList == 'all':
-            response = self.ixnObj.get(self.ixnObj.sessionUrl+'/vport')
-            vportList = ['%s' % vport['links'][0]['href'] for vport in response.json()]
-
-        # vportList: ['/api/v1/sessions/1/ixnetwork/vport/1', '/api/v1/sessions/1/ixnetwork/vport/2']
-        for eachVport in vportList:
-            self.ixnObj.patch(self.ixnObj.httpHeader+eachVport, data={'txMode':txMode})
+        pass
```
Comment (keysightgems) [inline Modules/IxNetRestApiPortMgmt.py:675]: Add implemented code here.

**Pull request closed without merging**